Apollo/Devices/Loop.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Apollo.DeviceViewers;
+using Apollo.Elements;
+using Apollo.Structures;
+
+namespace Apollo.Devices {
+    public class Loop: Device {
+        
+        struct LoopCourierInfo{
+            public int count, amount;
+            public Signal signal;
+            
+            public LoopCourierInfo(int c, int a, Signal s){
+                count = c;
+                amount = a;
+                signal = s;
+            }
+        }
+        Time _duration;
+        public Time Duration {
```

Comment:
If it has to do with timing / there is a Duration, there should be a Gate as well.